network/fan.go

```diff
@@ -70,6 +70,9 @@ func CalculateOverlaySegment(underlayCIDR string, fan FanConfigEntry) (*net.IPNe
 		newOverlaySize := overlaySize + (subnetSize - underlaySize)
 		fanSize := uint(underlaySize - overlaySize)
 		newFanIP := underlayNet.IP.To4()
+		if newFanIP == nil {
+			return nil, errors.New("fan address in not an IPv4 address.")
```

Comment:
'is not'
also, it is usually better for users if you include the content. so:
if newFanIP == nil {
  return nil, errors.Errorf("host address %v is not an IPv4 address, and is invalid for a FAN underlay address", underlayNet)

I do wonder whether we want a strict error, or just a

if newFanIP == nil { continue}

however, if we do that, we *definitely* don't want to add underlaySize into the newOverlaySize as it would break the computation.